### PR TITLE
Replace the zram service

### PR DESCRIPTION
**Pull request by poncovka**
Remove the Anaconda's implementation of the zram service and use the systemd
zram-generator instead. The inst.zram option is deprecated.

### CODE REVIEW

File: anaconda.spec.in [hunk 1]
```diff
@@ -184,6 +184,7 @@ Requires: tmux
 # install time crash handling
 Requires: gdb
 Requires: rsync
+Requires: zram-generator-defaults
```
Comment (poncovka) [inline anaconda.spec.in:187]: Do we want to have this dependency in `anaconda-install-env-deps`? Or should it be defined in the `lorax` templates and `spin-kickstarts` kickstart files?

File: data/liveinst/liveinst [hunk 2]
```diff
@@ -33,10 +33,6 @@ if [[ "$LIVECMD $*" =~ "--liveinst" ]]; then
     LIVE_INSTALL=1
 fi
 
-# Start swap on ZRAM service. Do this for LiveOS only, rather than enabling
-# it on the target system, for the time being.
-systemctl start zram.service
```
Comment (poncovka) [inline data/liveinst/liveinst:38]: Can we drop this?

File: data/systemd/anaconda-pre.service [hunk 3]
```diff
@@ -9,7 +9,6 @@ Wants=rsyslog.service
 Wants=systemd-udev-settle.service
 Wants=NetworkManager.service
 Wants=plymouth-quit.service plymouth-quit-wait.service
-Wants=zram.service
```
Comment (poncovka) [inline data/systemd/anaconda-pre.service:12]: Should the Anaconda services depend on `zram-generator`?

File: docs/boot-options.rst [hunk 4]
```diff
@@ -664,6 +664,7 @@ inst.zram
 ^^^^^^^^^
 
 Forces/disables (on/off) usage of zRAM swap for the installation process.
+See ``zram-generator`` for more information.
```
Comment (poncovka) [inline docs/boot-options.rst:667]: It depends on the issue https://github.com/systemd/zram-generator/issues/42.

File: anaconda.spec.in (hunk 1 above)
Comment (jkonecny12) [reply anaconda.spec.in:187]: Hmm, if we will deprecate the ``inst.zram`` boot option then I don't see any reason to have it as Anaconda dependency.

Comment (jkonecny12) [reply anaconda.spec.in:187]: Hmm, if we will deprecate the inst.zram boot option then I don't see any reason to have it as Anaconda dependency.

File: data/liveinst/liveinst (hunk 2 above)
Comment (jkonecny12) [reply data/liveinst/liveinst:38]: If Anaconda won't depend on this service than it should be handled by Lorax/Live creation.

File: data/systemd/anaconda-pre.service (hunk 3 above)
Comment (jkonecny12) [reply data/systemd/anaconda-pre.service:12]: The same answer as above.

File: docs/boot-options.rst (hunk 4 above)
Comment (poncovka) [reply docs/boot-options.rst]: The `inst.zram` option will be deprecated.

File: anaconda.spec.in (hunk 1 above)
Comment (poncovka) [reply anaconda.spec.in]: Changed to `Recommends`.

File: docs/boot-options.rst [hunk 5]
```diff
@@ -658,13 +658,6 @@ will be used by default, if found.
 
 See the |anacondalogging|_ for more info on setting up logging via virtio.
 
-.. inst.zram:
-
-inst.zram
-^^^^^^^^^
-
-Forces/disables (on/off) usage of zRAM swap for the installation process.
-
```
Comment (jkonecny12) [inline docs/boot-options.rst:667]: Please move this to the `Removed options` section.

Comment (poncovka) [reply docs/boot-options.rst:667]: Good point! Updated.

**Comment by poncovka** [comment]
Tested.

**Pull request merged**